README.md

```diff
@@ -54,16 +54,6 @@ A: Existing formats can continue to be a proving ground for technologies, as nee
 * June ??? v1.0.0
   * Release initial version of spec with two independent non-experimental implementations from OCI members
 
-## Maintainers
```

Comment:
I considered that but this README is already too long, and I think it's discoverable enough from the main repo landing page...